### DISCUSSION TITLE
version controlling .kenv

### DISCUSSION THREAD

**Discussion by kdabir** [johnlindquist/kit#216] [General]
I typically store dotfiles in version control so that on a new system, I have all it takes to make me productive. 

In case of ScriptKit, is it recommended to add .kenv to versions control. I think only `bin`, `scripts`, `.env`, `package.json` should suffice. Any suggestions?

**Comment by johnlindquist** [comment]
- `.env` I would never version control a `.env` file. Too many private keys. Scripts should prompt for missing vars anyway.
- `bin` is generated. It can be re-generated from the "Help-> Generate bin files". I may automate this a bit more in the future.
- `package.json` isn't technically necessary either. The scripts will prompt you to install missing packages

By default, I would recommend version controlling:
- scripts
- lib
- db

This would probably vary per user. In the future, you'll be able to switch between multiple `.kenv` environments on the same machine (e.g., one for work, one for personal), but I haven't started any work on that yet.

**Reply by kdabir** [reply]
thanks, this helps 👍
